packages/db-authorization/index.js

```diff
@@ -61,6 +61,39 @@ async function auth (app, opts) {
     }
   }
 
+  app.decorateRequest('authorize', endpointAuthorize)
+
+  async function endpointAuthorize () {
+    const request = this
+    const endpointRules = opts.rules.filter(rule => {
+      if (rule?.endpoint) {
+        const endPointRegEx = new RegExp(rule.endpoint)
+        return !!request.raw.url.match(endPointRegEx)
+      }
+      return false
+    })
```

Comment:
The whole point of adding a decorator is to support _not_ doing this in here. We want to provide a generic utility folks could use _manually_.

Then we use the data in `rule.endpoint` to automatically add a `preHandler` that call this endpoint with the given rule.

Reply:
I don't get the part about using it manually. Do you mean something like passing the rules to `req.authorize()` like @ivan-tymoshenko said?

It looks a bit confusing to me to be honest. I like the idea of using `preHandler` to prevent calling `req.authorize()` inside the plugin, but I'm not sure if it is possible if it's not using all the authorization rules from a configuration files

Reply:
I find this idea confusing too... @mcollina ..😉🙂